Add GitHub workflow for checking missing values schema changes

ubergesundheit opened the pull request:
This PR adds the already existing GitHub workflow to be templated using devctl.

I've changed the original to not use any external GitHub actions

This PR builds upon the excellent work of @glitchcrab done in https://github.com/giantswarm/template-app/pull/14

### Checklist

- [x] Update changelog in CHANGELOG.md.

glitchcrab reviewed (approved):
Oooo sexy

kopiczko commented:
Will that fail when only the default value is changed? E.g.:

```diff
- namespace: a
+ namespace: b
```

ubergesundheit commented:
Yes, you already found this scripts weak spot ;)

The workflow should check for changes:

- Renamed, removed, added keys
- type changes of unchanged keys
